File: examples/rest-api/python/services/ProductService.py
import os
import datetime
import boto3
dynamoDB = boto3.resource('dynamodb')


class ProductService:
    def __init__(self):
        self.dynamoDBClient = dynamoDB.Table(os.environ['PRODUCTS_TABLE_NAME'])

    def getItem(self, sku):
        return self.dynamoDBClient.get_item(
            Key={'sku': sku}
        )

    def createItem(self, sku, name, description):
        timestamp = datetime.datetime.utcnow().isoformat()

        item = {
            'sku': sku,
            'name': name,
            'description': description,
            'createdAt': timestamp,
            'updatedAt': timestamp
        }

        self.dynamoDBClient.put_item(Item=item, ReturnValues='NONE')

        return item

    def deleteItem(self, sku):

        return self.dynamoDBClient.delete_item(Key={'sku': sku})

    def updateItem(self, sku, name, description):

        return self.dynamoDBClient.update_item(
            Key={'sku':sku},
            ExpressionAttributeNames={
                '#name': 'name',
                '#description': 'description'
            },
            ExpressionAttributeValues={
                ':n': name,
                ':d': description
            },
            UpdateExpression="set #name = :n, #description = :d",
            ReturnValues="UPDATED_NEW"
        )
    
    def listItems(self):

        return self.dynamoDBClient.scan()

File: examples/rest-api/nodejs/src/services/ProductService.ts
import AWS from "aws-sdk";
import Product from "../models/Product";

class ProductService {
	db: any;
	tableName: string;

	constructor(tableName: string | undefined, db?: any) {
		this.db = db ? db : new AWS.DynamoDB.DocumentClient();
		this.tableName = tableName ? tableName : "";
	}

	getItem(key: string): Promise<any> {
		var params = {
			Key: {
				sku: key
			},
			TableName: this.tableName
		};

		return this.db.get(params).promise();
	}

	createItem(product: Product): Promise<any> {
		var params = {
			Item: product,
			TableName: this.tableName
		};

		return this.db.put(params).promise();
	}

	updateItem(product: Product): Promise<any> {
		var params = {
			ExpressionAttributeNames: {
				"#name": "name",
				"#description": "description"
			},
			ExpressionAttributeValues: {
				":name": product.name,
				":description": product.description
			},
			UpdateExpression: "SET #name = :name, #description = :description",
			Key: {
				sku: product.sku
			},
			ReturnValues: "ALL_NEW",
			TableName: this.tableName
		};

		return this.db.update(params).promise();
	}

	deleteItem(key: string): Promise<any> {
		var params = {
			Key: {
				sku: key
			},
			TableName: this.tableName
		};

		return this.db.delete(params).promise();
	}

	listItems(): Promise<any> {
		var params = {
			TableName: this.tableName
		};

		return this.db.scan(params).promise();
	}
}
export default ProductService;
